proofwatch/internal/metrics/observer_test.go

```diff
@@ -0,0 +1,247 @@
+package metrics
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/attribute"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+)
+
+// evidenceObserverTestFixture provides test infrastructure for EvidenceObserver tests
+type evidenceObserverTestFixture struct {
+	observer *EvidenceObserver
+	reader   *sdkmetric.ManualReader
+	t        *testing.T
+}
+
+// setupEvidenceObserverTest creates a test fixture with configured meter provider
+func setupEvidenceObserverTest(t *testing.T) *evidenceObserverTestFixture {
```

Comment:
In some test files the helper functions are defined at the end and others they are defined at the beginning. I don't have a strong opinion and know that technically nothing would change, but we should try to keep the style consistent among the files. This don't need to be done in this PR, but is something that we can gradually improve in separate PRs.